Tests/Microsoft.AppCenter.Test.Functional/Distribute/DistributeUpdateTest.cs

```diff
@@ -154,5 +155,129 @@ public async Task SetUpdateTrackPrivateTest()
+
+            // Start AppCenter.
+            AppCenter.UnsetInstance();
+            AppCenter.LogLevel = LogLevel.Verbose;
+            Distribute.UpdateTrack = updateTrack;
+
+            // Save update token.
+            if (updateTrack == UpdateTrack.Private)
+            {
+                DistributeEvent?.Invoke(this, DistributeTestType.SaveMockUpdateToken);
+            }
+            AppCenter.Start(Config.ResolveAppSecret(), typeof(Distribute));
+
+            // Wait for "startService" log to be sent.
+            await startServiceTask;
+            DistributeEvent?.Invoke(this, DistributeTestType.OnResumeActivity);
+
+            // Wait when Distribute will start.
+            await Distribute.IsEnabledAsync();
+
+            // Wait for processing event.
+            var result = await implicitCheckForUpdateTask;
+
+            // Verify response.
+            Assert.Equal("GET", result.Method);
+            Assert.True(result.Uri.Contains(urlDiff));
+            Assert.True(result.Uri.Contains(Config.ResolveAppSecret()));
+
+            // Check for update.
+            Distribute.CheckForUpdate();
+
+            // Wait for processing event.
+            result = await explicitCheckForUpdateTask;
+
+            // Verify response.
+            Assert.Equal("GET", result.Method);
+            Assert.True(result.Uri.Contains(urlDiff));
+            Assert.True(result.Uri.Contains(Config.ResolveAppSecret()));
+        }
+
+        [Fact]
+        public async Task DisableAuthomaticCheckUpdateTest()
+        {
+            // Enable Distribute for debuggable builds.
+            DistributeEvent?.Invoke(this, DistributeTestType.EnableDebuggableBuilds);
+
+            // Setup network adapter.
+            var httpNetworkAdapter = new HttpNetworkAdapter();
+            DependencyConfiguration.HttpNetworkAdapter = httpNetworkAdapter;
+            HttpResponse response = new HttpResponse()
+            {
+                Content = GetReleaseJson("3.0.0", "30", false),
+                StatusCode = 200
+            };
+            var explicitCheckForUpdateTask = httpNetworkAdapter.MockRequest(request => request.Method == "GET", response, 30);
+            var startServiceTask = httpNetworkAdapter.MockRequestByLogType("startService");
+
+            // Start AppCenter.
+            AppCenter.UnsetInstance();
+            AppCenter.LogLevel = LogLevel.Verbose;
+            Distribute.DisableAutomaticCheckForUpdate();
+            AppCenter.Start(Config.ResolveAppSecret(), typeof(Distribute));
+
+            // Wait for "startService" log to be sent.
+            await startServiceTask;
+            DistributeEvent?.Invoke(this, DistributeTestType.OnResumeActivity);
+
+            // Wait when Distribute will start.
+            await Distribute.IsEnabledAsync();
+
+            // Waitting any calls and verify that httpNetworkAdapter was never called.
```

Comment:
Comment and assert are confusing. need to be more accurate as in "we sent only the start service log" or something.